src/Statement/StatementList.php

```diff
@@ -325,4 +325,15 @@ public function filter( StatementFilter $filter ) {
 		return $statementList;
 	}
 
+	/**
+	 * @see http://php.net/manual/en/language.oop5.cloning.php
+	 *
+	 * @since 5.1
+	 */
+	public function __clone() {
+		foreach ( $this->statements as &$statement ) {
```

Comment:
It's really a shame we have to do that even in a context where no modification will ever happen.
Can we have an Immutable marker interface, so cloning can be skipped for anything that implements it?
That way, we could support mutable and immutable implementations of StatementList, TermList, etc.